packages/central-server/src/apiV2/utilities/translateExpression.js

```diff
@@ -22,7 +22,10 @@
 export const translateExpression = async (models, rawExpression, codes) => {
   const questionCodeToId = await models.question.findIdByCode(codes);
   let expression = rawExpression;
-  for (const code of codes) {
+
+  // We sort by string length (desc) so that we avoid replacing a substring in the regex
+  const codesOrderedByLength = codes.sort((a, b) => b.length - a.length);
```

Comment:
@IgorNadj kudos and appreciation for your insight here, my previous implementation was a lot more long-winded 😂